### DISCUSSION TITLE
fail2ban in a ubuntu container on 24.04 fix ?

### DISCUSSION THREAD

**Discussion by AndMck** [fail2ban/fail2ban#3686] [General]
hi I'm still new on linux and i'm trying to understand if there is a move to resolve the issue where fail2ban doesn't work with python 3.12 (i currently don't use systemd) , I've been trying to update a old pod built with debian buster to use ubuntu 24.04. I'm trying to understand whether i should continue with trying to resolve the vulnerabilities in 22.04 or wait/fix trying to get fail2ban working on 24.04. i do not use systemd but use supervisord as the process manager. maybe there is a better way to resolve my solution. this is being used as a sftp server.
the ERROR i'm getting is "No module named 'asynchat'"

**Comment by sebres** [comment]
1. Ubuntu 24.04 is not released yet, is it? 
However if I correct understand debian maintainers, they fixed it by manual injections with copy of `async*` modules to fail2ban.
Anyway I don't think it is simple task for a newbie, but you can try follow #3487 - the issue is fixed there and the solution is pretty explained.
2. Although it is fixed, it is not released yet (and no deb-file provided there), but you can try to [install f2b from source](https://github.com/fail2ban/fail2ban/wiki/How-to-install-or-upgrade-fail2ban-manually#manual-installation-from-source)

> i currently don't use systemd
> the ERROR i'm getting is "No module named 'asynchat'"

There is no relationship between systemd and support of python 3.12, in my opinion.
